app/models/pawn.rb

```diff
@@ -32,12 +66,14 @@ def black_pawn_two_step_move?(destination_x, destination_y)
 
 def white_pawn_diagonal_strike?(destination_x, destination_y)
   return true if color == 'white' &&
-                 !Piece.where(column_coordinate: destination_x, row_coordinate: destination_y).empty? &&
+                 game.pieces.where(column_coordinate: destination_x, row_coordinate: destination_y).present? &&
                  (diagonal_up_and_right_move?(destination_x, destination_y) || diagonal_up_and_left_move?(destination_x, destination_y))
+  false
 end
 
 def black_pawn_diagonal_strike?(destination_x, destination_y)
   return true if color == 'black' &&
-                 !Piece.where(column_coordinate: destination_x, row_coordinate: destination_y).empty? &&
+                 game.pieces.where(column_coordinate: destination_x, row_coordinate: destination_y).present? &&
```

Comment:
hmm Doesn't make sense. `exists?` is part of active record. If the result is `nil`, it returns false.